src/changefile/writeChangeFilesFixup.ts

```diff
@@ -0,0 +1,242 @@
+    if (changes.length > 1) {
+      console.warn(
+        'Fixup mode with individual change files only supports updating one package at a time. Using the first change.'
+      );
+    }
+    const newChange = changes[0];
+    const existingChange = existingChangeInfo as ChangeFileInfo;
+
+    // Merge comments and other properties
+    const mergedChange: ChangeFileInfo = {
+      ...existingChange,
+      ...newChange,
+      // Override with merged comment and higher change type
+      comment: existingChange.comment ? `${existingChange.comment}\n\n${newChange.comment}` : newChange.comment,
+      type: getHigherChangeType(existingChange.type, newChange.type),
+      // But preserve the original package name
+      packageName: existingChange.packageName,
+    };
+
+    fs.writeJSONSync(changeFilePath, mergedChange, { spaces: 2 });
+  }
+
+  // Stage the updated change file
+  stage([changeFilePath], cwd);
+
+  // Create a fixup commit using the original commit hash
+  console.log(`Original change file was added in commit: ${commitHash}`);
+
+  // Create a fixup commit
+  const commitResult = await gitAsync(['commit', '--fixup', commitHash], { cwd, verbose: true });
+
+  if (!commitResult.success) {
+    console.error(`Failed to create fixup commit: ${commitResult.errorMessage}`);
+    return null;
+  }
+
+  console.log(`Created fixup commit for ${commitHash}`);
+  console.log(`Updated change file: ${changeFilePath}`);
+
+  return changeFilePath;
+}
+
+const MAX_COMMITS = 100;
+
+/**
+ * Finds the most recent commit that introduced change files by searching through
+ * the current branch's commits from newest to oldest.
+ * @param cwd - Working directory
+ * @param changePath - Path to the change directory
+ * @returns Object with commit hash and change file name, or null if none found
+ */
+async function findMostRecentChangeFileCommit(
+  cwd: string,
+  changePath: string
+): Promise<{ commitHash: string; changeFileName: string } | null> {
+  const relativeChangePath = path.relative(cwd, changePath);
+
+  // Get the fork point (merge base with main/master branch)
+  const forkPointResult = await findForkPoint(cwd);
+  if (!forkPointResult) {
+    console.warn('Unable to determine fork-point for current branch');
+    return null;
+  }
+
+  // Get commits on the current branch, limited to MAX_COMMITS
+  const logResult = await gitAsync(
+    ['log', '--oneline', '--format=%H', `-n`, MAX_COMMITS.toString(), `${forkPointResult}..HEAD`],
+    { cwd }
+  );
+  if (!logResult.success) {
+    console.warn('Failed to get git log for finding recent change file commits');
+    return null;
+  }
+
+  const commits = logResult.stdout
+    .trim()
+    .split('\n')
+    .filter(line => line.trim());
+
+  // Search through commits from newest to oldest
+  for (const commitHash of commits) {
+    // Check what files were added in this commit (compare with parent)
+    const diffResult = await gitAsync(
```

Comment:
There's already some closely related logic in `getChangedPackages`, so it would be best if at least some of that could be reused/combined with the logic here.